consumerOptions.go

```diff
@@ -21,51 +21,54 @@
 package kafkaclient
 
 import (
-	"github.com/Shopify/sarama"
-	"github.com/bsm/sarama-cluster"
 	"github.com/uber-go/kafka-client/internal/consumer"
-	"time"
 )
 
-// ConsumerOption is the type for optional arguments to the NewConsumer constructor.
 type (
+	// ConsumerOption is the type for optional arguments to the NewConsumer constructor.
 	ConsumerOption interface {
 		apply(*consumer.Options)
 	}
-	consumerLimitOption struct {
-		limits map[consumer.TopicPartition]int64
+
+	consumerBuildOptions []ConsumerOption
+
+	partialConstruction struct {
+		enabled bool
+		errs    *consumerErrorList
 	}
 )
 
-var defaultOptions = consumer.Options{
-	Concurrency:            1024,
-	RcvBufferSize:          2 * 1024, // twice the concurrency for compute/io overlap
-	PartitionRcvBufferSize: 32,
-	OffsetCommitInterval:   time.Second,
-	RebalanceDwellTime:     time.Second,
-	MaxProcessingTime:      250 * time.Millisecond,
-	OffsetPolicy:           sarama.OffsetOldest,
-	ConsumerMode:           cluster.ConsumerModePartitions,
-	Limits:                 nil,
+// PartialConstructionError returns a list of topics that could not be consumed as a list of ConsumerError.
+// PartialConstructionError should be called on the ConsumerOption returned by EnablePartitionConstruction.
+// This is useful if you chose to EnablePartialConstruction.
+func PartialConstructionError(option ConsumerOption) []ConsumerError {
```

Comment:
this is very weird. How about we punt on exposing methods that return builder errors for now ? Lets just log the errors for now and be done with it. Let's tackle the exposing the error part separately, we need to put more thought into this. Meanwhile, would love to get this multiClusterConsumer landed

Reply:
Fair. Punted.